src/util/path/index.ts

```diff
@@ -755,81 +866,76 @@ export const getPointOnPath = (
  * ];
  *
  */
-export const parsePath = (pathString) => {
-  // one of commands (m,M,l,L,q,Q,c,C,etc.) followed by non-command characters (i.e. command values)
-  const re = rePathCommand,
-    rNumber = '[-+]?(?:\\d*\\.\\d+|\\d+\\.?)(?:[eE][-+]?\\d+)?\\s*',
-    rNumberCommaWsp = `(${rNumber})${commaWsp}`,
-    rFlagCommaWsp = `([01])${commaWsp}?`,
-    rArcSeq = `${rNumberCommaWsp}?${rNumberCommaWsp}?${rNumberCommaWsp}${rFlagCommaWsp}${rFlagCommaWsp}${rNumberCommaWsp}?(${rNumber})`,
-    regArcArgumentSequence = new RegExp(rArcSeq, 'g'),
-    result = [];
-
-  if (!pathString || !pathString.match) {
-    return result;
-  }
-  const path = pathString.match(/[mzlhvcsqta][^mzlhvcsqta]*/gi);
-
-  for (let i = 0, len = path.length; i < len; i++) {
-    const currentPath = path[i];
-    const coordsStr = currentPath.slice(1).trim();
-    const coords = [];
-    let command = currentPath.charAt(0);
-    const coordsParsed = [command];
-
-    if (command.toLowerCase() === 'a') {
-      // arcs have special flags that apparently don't require spaces so handle special
-      for (let args; (args = regArcArgumentSequence.exec(coordsStr)); ) {
-        for (let j = 1; j < args.length; j++) {
-          coords.push(args[j]);
-        }
-      }
-    } else {
-      let match;
-      while ((match = re.exec(coordsStr))) {
-        coords.push(match[0]);
-      }
-    }
+export const parsePath = (pathString: string): TComplexPathData => {
+  // clean the string
+  // add spaces around the numbers
+  pathString = pathString
+    .replace(new RegExp(`(${numberRegExStr})`, 'gi'), ' $1 ')
+    // replace annoying commas and arbitrary whitespace with single spaces
+    .replace(/,/gi, ' ')
+    .replace(/\s+/gi, ' ');
 
-    for (let j = 0, jlen = coords.length; j < jlen; j++) {
-      const parsed = parseFloat(coords[j]);
-      if (!isNaN(parsed)) {
-        coordsParsed.push(parsed);
+  const res: TComplexPathData = [];
+  for (const match of pathString.matchAll(rePathCommand)) {
+    let matchStr = match[0];
+    const chain: TComplexPathData = [];
+    let paramArr: RegExpExecArray | null;
+    do {
+      paramArr = new RegExp(rePathCommand.source, 'id').exec(matchStr);
```

Comment:
ok let's focus on this code for a second.
match and paramArr are always identical to me.
Can you explai why we regex twice?

Reply:
on line 909 we change `matchStr` by removing the command we just found. This is to account for chained commands.

Reply:
oh OK because `L 10 23 23 32 32 32 12 32 32 12 32 12 32 21 32 112 32` it gets parsed initially as a single command

Reply:
Correct. I'm going to resolve these convos, unreslove them in necessary